### PR TITLE
gorilla perf - inline some functions

### PR DESCRIPTION
**Pull request by OfirMos**

### CODE REVIEW

**Review by danni-m — APPROVED** [review]

**Comment by filipecosta90** [comment]
@OfirMos @danni-m inlining sometimes might be counterproductive. I believe in this case it did helped improve performance by 1-2% as shown below (used command=`redisbench-admin compare --github_repo RedisTimeSeries --github_org RedisTimeSeries --triggering_env circleci --baseline-branch master --comparison-branch perf_gorilla_inline --metric_name Totals.overallQueryRates.all_queries`) :

### Comparison between master and perf_gorilla_inline for metric: Totals.overallQueryRates.all_queries. Time Period from 7 days ago to a second ago
|             Test Case             |Baseline (median obs. +- std.dev)|Comparison (median obs. +- std.dev)|% change (higher-better)|            Note             |
|-----------------------------------|---------------------------------|-----------------------------------|------------------------|-----------------------------|
|tsbs-scale100_cpu-max-all-1        | 1553.641 +- 3.5%                | 1611.913 +- nan%                  |3.8%                    |potential IMPROVEMENT        |
|tsbs-scale100_cpu-max-all-8        | 209.334 +- 3.1%                 | 216.870 +- nan%                   |3.6%                    |potential IMPROVEMENT        |
|tsbs-scale100_double-groupby-1     | 111.677 +- 1.8%                 | 112.543 +- nan%                   |0.8%                    |potential IMPROVEMENT        |
|tsbs-scale100_double-groupby-5     | 22.496 +- 2.2%                  | 22.893 +- nan%                    |1.8%                    |potential IMPROVEMENT        |
|tsbs-scale100_double-groupby-all   | 11.544 +- 2.0%                  | 11.681 +- nan%                    |1.2%                    |potential IMPROVEMENT        |
|tsbs-scale100_groupby-orderby-limit| 1336.550 +- 12.4% UNSTABLE      | 1427.600 +- nan%                  |6.8%                    |UNSTABLE (very high variance)|
|tsbs-scale100_high-cpu-1           | 226.505 +- 4.1%                 | 228.280 +- nan%                   |0.8%                    |potential IMPROVEMENT        |
|tsbs-scale100_high-cpu-all         | 3.781 +- 2.7%                   | 3.235 +- nan%                     |-14.4%                  |REGRESSION                   |
|tsbs-scale100_lastpoint            | 537.310 +- 2.7%                 | 545.348 +- nan%                   |1.5%                    |potential IMPROVEMENT        |
|tsbs-scale100_single-groupby-1-1-1 | 20740.336 +- 13.1% UNSTABLE     | 21291.779 +- nan%                 |2.7%                    |UNSTABLE (very high variance)|
|tsbs-scale100_single-groupby-1-1-12| 4051.711 +- 2.5%                | 4146.115 +- nan%                  |2.3%                    |potential IMPROVEMENT        |
|tsbs-scale100_single-groupby-1-8-1 | 2457.688 +- 2.1%                | 2555.902 +- nan%                  |4.0%                    |potential IMPROVEMENT        |
|tsbs-scale100_single-groupby-5-1-1 | 1428.585 +- 1.6%                | 1469.538 +- nan%                  |2.9%                    |potential IMPROVEMENT        |
|tsbs-scale100_single-groupby-5-1-12| 433.954 +- 2.1%                 | 444.262 +- nan%                   |2.4%                    |potential IMPROVEMENT        |
|tsbs-scale100_single-groupby-5-8-1 | 709.544 +- 2.4%                 | 734.541 +- nan%                   |3.5%                    |potential IMPROVEMENT        |

But we should have double checked the `tsbs-scale100_high-cpu-all` prior merge:
image

TLDR, let's always associate to a perf PR the `redisbench-admin compare table`. Noneteless good job on this! ( except for the rush on merging! :) )

**Comment by filipecosta90** [comment]
@OfirMos @danni-m we might need to revert this PR. 
```
$ redisbench-admin compare --github_repo RedisTimeSeries --github_org RedisTimeSeries --triggering_env circleci --baseline-branch master --comparison-branch perf_gorilla_inline --metric_name Totals.overallQueryRates.all_queries --deployment_type oss-cluster --deployment_name oss-cluster-30-primaries --from_timestamp 1641715200286 --metric_mode 'lower-better' 
WARNING:root:Unable to fill git vars. caught the following error: 
Effective log level set to INFO
INFO:root:Using: redisbench-admin 0.6.8
2022-01-17 11:48:12,457 INFO Using: redisbench-admin 0.6.8
INFO:root:Checking connection to RedisTimeSeries with user: None, host: benchmarks.redislabs.com, port: 12000
2022-01-17 11:48:12,457 INFO Checking connection to RedisTimeSeries with user: None, host: benchmarks.redislabs.com, port: 12000
INFO:root:Using deployment_type=oss-cluster and deployment_name=oss-cluster-30-primaries for the analysis
2022-01-17 11:48:13,055 INFO Using deployment_type=oss-cluster and deployment_name=oss-cluster-30-primaries for the analysis
INFO:root:Using a time-delta from 8 days ago to now
2022-01-17 11:48:13,056 INFO Using a time-delta from 8 days ago to now
WARNING:root:Based on test-cases set (key=ci.benchmarks.redislabs/circleci/RedisTimeSeries/RedisTimeSeries:testcases) we have 49 distinct benchmarks. 
2022-01-17 11:48:13,244 WARNING Based on test-cases set (key=ci.benchmarks.redislabs/circleci/RedisTimeSeries/RedisTimeSeries:testcases) we have 49 distinct benchmarks. 
INFO:root:Printing differential analysis between branches
2022-01-17 11:48:22,886 INFO Printing differential analysis between branches
# Comparison between master and perf_gorilla_inline for metric: Totals.overallQueryRates.all_queries. Time Period from 8 days ago to now
|                 Test Case                 |Baseline (median obs. +- std.dev)|Comparison (median obs. +- std.dev)|% change (lower-better)|            Note             |
|-------------------------------------------|---------------------------------|-----------------------------------|-----------------------|-----------------------------|
|tsbs-scale100-cpu-max-all-1@4139rps        | 2039.311 +- 4.2%                | 2006.300 +- nan%                  |1.6%                   |potential IMPROVEMENT        |
|tsbs-scale100-cpu-max-all-8@588rps         | 588.201 +- 0.0%                 | 587.795 +- nan%                   |0.1%                   |potential IMPROVEMENT        |
|tsbs-scale100-double-groupby-1@324rps      | 324.374 +- 0.1%                 | 324.178 +- nan%                   |0.1%                   |potential IMPROVEMENT        |
|tsbs-scale100-double-groupby-5@70rps       | 70.111 +- 0.1%                  | 70.077 +- nan%                    |0.0%                   |potential IMPROVEMENT        |
|tsbs-scale100-double-groupby-all@35rps     | 35.149 +- 0.1%                  | 35.120 +- nan%                    |0.1%                   |potential IMPROVEMENT        |
|tsbs-scale100-groupby-orderby-limit@28rps  | 28.355 +- 0.0%                  | 28.355 +- nan%                    |-0.0%                  |potential REGRESSION         |
|tsbs-scale100-high-cpu-1@1816rps           | 176.677 +- 66.6% UNSTABLE       | 407.105 +- nan%                   |-56.6%                 |UNSTABLE (very high variance)|
|tsbs-scale100-high-cpu-all@18rps           | 18.062 +- 0.5%                  | 18.037 +- nan%                    |0.1%                   |potential IMPROVEMENT        |
|tsbs-scale100-lastpoint@488rps             | 488.089 +- 0.1%                 | 488.937 +- nan%                   |-0.2%                  |potential REGRESSION         |
|tsbs-scale100-single-groupby-1-1-12@2320rps| 2314.223 +- 2.7%                | 2314.386 +- nan%                  |-0.0%                  |potential REGRESSION         |
|tsbs-scale100-single-groupby-1-1-1@2320rps | 2317.145 +- 0.9%                | 2315.341 +- nan%                  |0.1%                   |potential IMPROVEMENT        |
|tsbs-scale100-single-groupby-1-8-1@4458rps | 2177.220 +- 4.1%                | 2153.518 +- nan%                  |1.1%                   |potential IMPROVEMENT        |
|tsbs-scale100-single-groupby-5-1-12@648rps | 648.101 +- 0.1%                 | 648.949 +- nan%                   |-0.1%                  |potential REGRESSION         |
|tsbs-scale100-single-groupby-5-1-1@648rps  | 648.149 +- 0.1%                 | 649.426 +- nan%                   |-0.2%                  |potential REGRESSION         |
|tsbs-scale100-single-groupby-5-8-1@1158rps | 1158.014 +- 0.0%                | 1157.624 +- nan%                  |0.0%                   |potential IMPROVEMENT        |
INFO:root:Detected a total of 14 stable tests between versions.
2022-01-17 11:48:22,967 INFO Detected a total of 14 stable tests between versions.
WARNING:root:Detected a total of 1 highly unstable benchmarks.
2022-01-17 11:48:22,967 WARNING Detected a total of 1 highly unstable benchmarks.
```
- It affects the high-cpu-1 query  in a large factor. 
- We need to decide either keep the PR and optimize the query or revert it given the extra 3-5% improvement on the other queries might not compensate for the extra 50% penalty on this one. 
The query is of type:
```
'
  use case: tsbs devops scale 100 use-case
  query: high-cpu-1
  tsbs query detail: All the readings where one metric is above a threshold for a particular host
  sample query:
    1st get the series names that pass the condition:
      "TS.MRANGE" "1451729628728" "1451772828728" "FILTER_BY_VALUE" "90.0" "1000" "FILTER" "fieldname=usage_user" "measurement=cpu" "hostname=host_1" "GROUPBY" "fieldname" "REDUCE" "max"
    then pass the given hostnames and timestamps to a second query:
      "TS.MRANGE" "1451656883002" "1451700083002" "FILTER_BY_TS" "1451665600000" "1451665620000" "1451665630000" "1451665640000" (...)
  '
```

```
